### DISCUSSION TITLE
What's next for Material Web Components in 2024

### DISCUSSION THREAD

**Discussion by asyncliz** [material-components/material-web#5402] [Announcements]
**tl;dr our [roadmap](https://github.com/material-components/material-web/blob/main/docs/roadmap.md) is updated for 2024!**

**This year we are working on accessibility, TSX support, expanded tokens, and three new components.**

<br>

Happy new year everyone! 🥳

2023 was a big year for us: we launched [@material/web](https://npmjs.com/package/@material/web) 1.0 with 19 production-ready components. It was quite the sprint involving a full rewrite AND design system overhaul for Material 3! 😱

This year post-release, we're focusing on sustainable, helpful, and realistic goals. Here's what's coming next!

- **Accessibility** - Adhere to best practices and support more screen readers.
- **TSX support** - Improve the experience of using MWC with [React](https://react.dev/).
- **Expanded tokens** - Make it easier to use tokens, like typography, in your own components; add shape and motion tokens for expressive design.
- **New components** - Adding card, snackbar, and navigation drawer.

Check out our updated [roadmap](https://github.com/material-components/material-web/blob/main/docs/roadmap.md) for all the details! (there's some juicy bits on prototyping Material layouts 👀)

Thank you all for your continued feedback and support!

❤️ The Material Web team

**Comment by tve** [comment]
> TSX support - Improve the experience of using MWC with [React](https://react.dev/).

I just started using MWC with Vue 3 (and Vite) and so far it's a breeze. "It just works". I hope that improving the experience with React doesn't detract from this. The biggest issue I've run into is that the material-color-utilities are not up-to-date in what they produce. Also, the API docs are missing the slot information (I saw a ticket open on that). Something that would be great is to start a "using MWC with Vue" page and let the community provide additions (or perhaps this should start as a discussion thread).

**Comment by Daydreamer-riri** [comment]
Great! Thank you for your efforts.

**Comment by aszenz** [comment]
I would really a focus on using web components as replacement for traditional form elements rendered in server side frameworks like django.

Also importing individual components as esm would be nice to have.

**Comment by brandonsbque** [comment]
> TSX support - Improve the experience of using MWC with [React](https://react.dev/).

Hey guys! really love the components so far! I made a python script that will generate a file that other devs can paste into the root folder of their React project, and it would make all the components in @material/web available to use in React. This is all based on a workaround that one of the actual devs for @material/web suggested found [here](https://github.com/material-components/material-web/discussions/5186). I tried my best to write a read.me for it that would help users on how to use the script, and how to actually use the @material/web components in React.

Here's the repo! https://github.com/brandonsbque/custom-component-gen . Hopefully, I'm not breaking any Github or you guys' policy here by linking my repo, since this is the first time i've done something like this, but if so please let me know! I hope this helps others 😄 I would really appreciate it if you guys gave some stars on the repo too if you found it helpful 🤩

**Comment by treeder** [comment]
Does this mean we won't see the rest of the components until 2025?

**Reply by asyncliz** [reply]
That's correct 🥲

The unfortunate reality is we're too small of a team to both build new components and provide the support for folks to use them, so balance is needed.

**Reply by treeder** [reply]
@asyncLiz does your team have some funds to sponsor some devs on GitHub to help? @floinay is right.

**Reply by asyncliz** [reply]
I wish :(

**Reply by christophe-g** [reply]
@asyncLiz - candidly asking if there something the community can try to do to help the material team have some more support ? 

The work here is stellar and what is built is really important. 

On the other hand it is kind of frustrating to see a lack corporate support for this project.

**Reply by Daydreamer-riri** [reply]
> @asyncLiz - candidly asking if there something the community can try to do to help the material team have some more support ? 
> 
> The work here is stellar and what is built is really important. 
> 
> On the other hand it is kind of frustrating to see a lack corporate support for this project.

I think so too. I would love to contribute to this project.

**Reply by asyncliz** [reply]
@christophe-g thank you, I'm frustrated as well 😞

I know just commenting can feel helpless, but we need more people to make comments like yours that show why we need to support this library and how not doing so impacts Google's brand identity.

**Reply by asyncliz** [reply]
Also, for anyone who wants to help out, start by [joining the lit.dev/discord #material channel](https://discord.com/invite/ysNzrXQn).

We can talk more there :)

**Reply by Danielnewheart** [reply]
Hi @asyncLiz Thanks for your effort. I really like this project and I believe google are connecting design and development. I would like to contribute but I cannot join your discord by the link you provide 🥲

**Reply by asyncliz** [reply]
@Danielnewheart try https://lit.dev/discord

**Comment by floinay** [comment]
too slow

**Reply by MarkStega** [reply]
@asyncLiz 

Several years ago my company needed to migrate a SilverLight application to something supported and modern. I faced a lot of controversy advocating use of Material Design (hereinafter called MW2). I finally won and proceeded to implement the app and in the process helped author Material.Blazor which brought MW2 to Blazor components.

As all software does, the aging of the MW2 implementation was beginning to show.

Late last year I pitched Material Web Components 3 (MWC3) as a path forward. Material Design made a huge positive leap and it appeared that MWC3 was shaping up to be an excellent path forward. MWC3 is lightweight, easy to use, absolutely gorgeous in both the styling and animations. We jumped in and began an implementation of Material.Blazor.MD3. 

MWC3 was a strong indicator (in our opinion) of the commitment of Google to Material Design 3 and a professional  implementation of web components.

It is gut wrenching to see the project now on death's doorstop (reminds me of the Spamalot cry of "I'm not dead, yet!"). The roadmap basically admits that the resource constraints effectively make the project moribund.

I am basically back to looking at commercial alternatives (Like the Telerik packages).

Although my opinion won't sway anyone, I hate to see Google act like Microsoft, pushing a design paradigm, getting developers excited, and then pulling the rug out from under. I now equate Google's MD3/MWC3 with Microsoft's phone foray.

Sorry for the rant, but I at least feel better putting this in writing.

I took the advice and have material channel in discord followed.

**Reply by simonziegler** [reply]
I'm Mark's co-developer of Material.Blazor, and Material 2 has served my startup excellently for the four years we have been developing Material.Blazor. My clients are financial institutions and Material's subtle clues to action aid the UX greatly.

But Material 2 is now showing its age, and the components start badly on pages with many components. Your work with Material 3 web components has been outstanding so far and would certainly resolve this issue.

Like Mark, I'm deeply disappointed to find that this project seems all but dead. The components you built so far are amazing, but they don't comprise a full suite.

So please, please resource this project to the extent it deserves. In the grand scheme of Google's operation this is a very low cost way of maintaining good relations with the developer community.

**Reply by tve** [reply]
I can echo what has been said by Mark and Simon. I've been looking for a very lightweight set of basic components and MWC3 is the best I've found, plus what you have works very well. From the 1.0 release I assumed this project had some momentum but the roadmap makes it clear that it doesn't. At this rate I'll have to scrap what I have build on MWC3 and look for something else.

**Reply by christophe-g** [reply]
I started drafting a short note aimed at being posted as a new discussion on material web (re material 2024 ambitions and beyond). 

The idea would be that it could have more impact if more than one individual write/post it. I also wish to collect other aligned views (great quality of the current components, but not enough corporate attention) to give it more strength. 

Also, I am not great at writing, so contributions are welcome ; ) 

It is here: https://semestriel.framapad.org/p/9dr9tcfd9x-a5y6?lang=en

**Reply by Daydreamer-riri** [reply]
This is an excellent idea! I'm even worse at writing and was wondering if there was something I could do to show my support?

**Reply by lkraav** [reply]
Those looking for a full suite of top quality Nordic engineering open source components might want to check out https://github.com/vaadin/web-components

Waiting for the unknown here is a thankless game.

**Reply by christophe-g** [reply]
Indeed vaadin components are great - the grid in particular. Even though lit-based of vaadin are under an experimental flag, they still rely on Polymer.  Ideally, I'd wish to only have a single templating dependency: `lit`.

**Reply by lkraav** [reply]
> Indeed vaadin components are great - the grid in particular. Even though lit-based of vaadin are under an experimental flag, they still rely on Polymer.  Ideally, I'd wish to only have a single templating dependency: `lit`. 

Yes, Lit is in experimental phase, but majority of the components have pure Lit version available by now. I got my company to back some tens of hrs of this conversion work for cxl.com get in touch with Vaadin if you have the ability to join backers!

**Reply by nedrise27** [reply]
> Those looking for a full suite of top quality Nordic engineering open source components might want to check out
> https://github.com/vaadin/web-components
> 
> Waiting for the unknown here is a thankless game.

Does it have a license?

**Reply by treeder** [reply]
Hi all, since I have several projects that depend on these components and I can't wait for this project to start moving again, I forked it so my team and I can continue to develop it. 

You can find it here for now: https://github.com/treeder/material

And you can start using it today with all the exact same components, see README.

**Reply by mindaugasvcs** [reply]
> Hi all, since I have several projects that depend on these components and I can't wait for this project to start moving again, I forked it so my team and I can continue to develop it.
> 
> You can find it here for now: https://github.com/treeder/material
> 
> And you can start using it today with all the exact same components, see README.

Looks and feels awful

**Reply by treeder** [reply]
What do you mean @mindaugasvcs ? It's literally identical, what a silly comment.

**Reply by asyncliz** [reply]
Please remember to [be respectful and follow our code of conduct](https://github.com/material-components/material-web/blob/main/CODE_OF_CONDUCT.md#our-standards). Thanks :)

**Reply by mindaugasvcs** [reply]
> What do you mean @mindaugasvcs ? It's literally identical, what a silly comment.

![Untitled](https://github.com/user-attachments/assets/857f4133-bbb3-4c76-a531-1872ddb2f2a6)

**Reply by treeder** [reply]
Use version 0.X, [1.X is beta](https://github.com/treeder/material/discussions/31) and will have some bugs. 

You see one little issue and it looks and feels awful huh? @mindaugasvcs please stick to this repo and do not use the one I mentioned, you are not welcome there.

**Comment by KyleRicardo** [comment]
With all due respect, is it possible that you have not completed the development of MD3 yet and MD4 has already been released.

**Reply by peschee** [reply]
There is no such thing as MD4 afaik.

**Comment by philippecgie** [comment]
I don't agree with the roadmap; why focus on accessibility, expanded tokens, ... when essential components like navigation drawer are not yet implemented? Without these, nobody can use the library...

Also, no need to focus on experience  improvement with react, they have great libraries on their own. I recommend focusing on vanilla JS usage

**Reply by treeder** [reply]
Totally agree. I'm already moving away from this library based on this roadmap, none of those things help me. And 100% on the react comment, they don't need yet another library.

**Reply by anlexN** [reply]
Totally agree. This is roadmap's problem. very silly problem.

**Comment by dreamstar-enterprises** [comment]
Thanks for all the great efforts!

Any plans or supporting Material 3 Web Components for Angular too? E.g. Md-checkbox vs the older Mat-Checkbox, with all the new Material 3 customisations?

**Reply by floinay** [reply]
https://material.angular.io/guide/material-3

**Comment by Paria-Heidari** [comment]
@asyncLiz 
Hi, :)
Could you please let me know when the carousel component will be available for implementation in the web environment?
Thanks!

**Comment by phuzi** [comment]
Meanwhile in 2025 I'm still stuck on Material 2 as [Navigation Drawer](https://m3.material.io/components/navigation-drawer/overview#b56dddd7-33c7-4297-8750-5b86833f292e) and [Cards](https://m3.material.io/components/cards/overview#ffeb6746-b6eb-4715-ad11-4a93f9e221ec) still appear to be "planned"!

**Reply by treeder** [reply]
Card and nav drawer are in this fork: https://github.com/treeder/material

**Reply by agperegrina** [reply]
> Card and nav drawer are in this fork: https://github.com/treeder/material

Why don't you contribute them to MWC? It would be really good to avoid having to install MWC from several sources...

**Reply by treeder** [reply]
Because this project is dead. And you don't need to install from several sources, just pick one.

**Reply by neoacevedo** [reply]
> > Card and nav drawer are in this fork: https://github.com/treeder/material
> 
> Why don't you contribute them to MWC? It would be really good to avoid having to install MWC from several sources...

https://github.com/material-components/material-web/discussions/5642 

This is the main reason to everything.

**Comment by xHuSKeRx** [comment]
Cool
